routers/api/v1/repo/pull.go

```diff
@@ -497,7 +498,37 @@ func MergePullRequest(ctx *context.APIContext) {
 		return
 	}
 
-	if err := pr.Merge(ctx.User, ctx.Repo.GitRepo); err != nil {
+	if len(form.Do) == 0 {
+		form.Do = "merge"
+	}
+
+	if models.MergeStyle(form.Do) != models.MergeStyleRegular &&
```

Comment:
nit: This is check unnecessary, since `pr.Merge` already checks the provided `mergeStyle`.